scripts/release.sh

```diff
@@ -0,0 +1,200 @@
+            writeError $error
+            exit -2
+        fi
+    ;;
+    "branch")
+        # Quality releases should run from a release branch
+        if [[ $branch_name =~ "release-" ]]; then
+            NEW_BRANCH_VERSION="${MAJOR}.$(( MINOR + 1 ))"
+            NEW_BRANCH_NAME="release-${NEW_BRANCH_VERSION}"
+            msg="Doing a quality branch: ${NEW_BRANCH_NAME}"
+            writeInfo $msg
+
+            if git show-ref --verify --quiet refs/heads/${NEW_BRANCH_NAME}; then
+                error="branch ${NEW_BRANCH_NAME} exists"
+                writeError $error
+                exit -3
+            fi
+
+            NEW_PKG_VERSION="${NEW_BRANCH_VERSION}.0-rc0"
+            git checkout -b "${NEW_BRANCH_NAME}"
+            writePackageVersion "${NEW_PKG_VERSION}"
+            tagDescription="Quality branch"
+            tag "${NEW_PKG_VERSION}" "${tagDescription}"
+            msg="locally created quality branch. In order to build you'll have to"
+            writeInfo $msg
+            echo "$ git push --follow-tags ${gitOrigin} ${NEW_BRANCH_NAME}:${NEW_BRANCH_NAME}"
+
+        else
+            if [[ $branch_name != "master" ]]; then
+                msg="You are branching on ${branch_name} instead of master or a relase-branch"
+                writeWarning $msg
+                read -p "Do you wish to continue? [y/n]" -n 1 -r
+                if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+                    exit 1
+                fi
+            fi
+            NEW_BRANCH_VERSION="${MAJOR}.${MINOR}"
+            NEW_BRANCH_NAME="release-${NEW_BRANCH_VERSION}"
+            NEW_PKG_VERSION="${NEW_BRANCH_VERSION}.0-rc0"
+            MASTER_PKG_VERSION="${MAJOR}.$(( MINOR + 2 )).0-develop"
+            msg="Creating a new features branch: ${NEW_BRANCH_NAME}"
+            writeInfo $msg
+
+            if git show-ref --verify --quiet refs/heads/${NEW_BRANCH_NAME}; then
+                error="branch ${NEW_BRANCH_NAME} exists"
+                writeError $error
+                exit -3
+            fi
+
+            git branch "${NEW_BRANCH_NAME}"
+            msg="Writing new package version for development: ${MASTER_PKG_VERSION}"
+            writeInfo $msg
+            writePackageVersion "${MASTER_PKG_VERSION}"
+            git checkout "${NEW_BRANCH_NAME}"
+            writePackageVersion "${NEW_PKG_VERSION}"
+            tagDescription="NewFeatures branch"
+            tag "${NEW_PKG_VERSION}" "${tagDescription}"
+            msg="Locally created new features branch. In order to build you'll have to"
+            writeInfo $msg
+            echo "$ git push --follow-tags ${gitOrigin} ${NEW_BRANCH_NAME}:${NEW_BRANCH_NAME}"
+            echo "For writing master changes you'll need to:"
+            echo "$ git push ${gitOrigin} ${branch_name}:${branch_name}"
+
+        fi
+    ;;
+    *)
+        writeError "Only branch|rc|final parameters are accepted"
+        exit -1
+    ;;
+esac
```

Comment:
* Please make sure to don't mix camel case and all underscore var names (the Bashism is to use underscore instead)
* Please make sure to quote all the strings
* Please make sure to use curly brackets in var name to avoid incorrect expansion `$hello` -> `${hello}`